back-end/src/sciler/config/configHandler.go

```diff
@@ -53,23 +53,84 @@ func generateDataStructures(readConfig ReadConfig) (WorkingConfig, error) {
 func checkConfig(config WorkingConfig) error {
 	for _, puzzle := range config.Puzzles {
 		for _, rule := range puzzle.Event.Rules {
-			err := rule.Conditions.CheckConstraints(config)
-			if err != nil {
+			if err := rule.Conditions.checkConstraints(config); err != nil {
+				return err
+			}
+			if err := checkActions(rule.Actions, config); err != nil {
 				return err
 			}
 		}
 	}
 
 	for _, generalEvent := range config.GeneralEvents {
 		for _, rule := range generalEvent.Rules {
-			err := rule.Conditions.CheckConstraints(config)
-			if err != nil {
+			if err := rule.Conditions.checkConstraints(config); err != nil {
+				return err
+			}
+			if err := checkActions(rule.Actions, config); err != nil {
 				return err
 			}
 		}
 	}
+	return nil
+}
 
-	// todo add check for actions
+// checkAction is a method that will return an error is the actions value types and instructions are not equal to the device output specifications
+func checkActions(actions []Action, config WorkingConfig) error {
+	for _, action := range actions {
+		switch action.Type {
+		case "device":
+			{
+				if device, ok := config.Devices[action.TypeID]; ok { // checks if device can be found in the map, if so, it is stored in variable device
```

Comment:
can you make this a separate mehtod? this is not very overzichtelijk

Reply:
Done